lib/Certificate.php

```diff
@@ -38,7 +38,7 @@ public static function pemToDer($pem)
 
     public function __construct($pem)
     {
-        if (\is_string($pem)) {
+        if (\is_string($pem) || $pem instanceof \OpenSSLCertificate) {
```

Comment:
This should rather go into the `else` branch where `if (\is_resource($pem)) {` is checked, no?

Reply:
for PHP 8 openssl_x509_read accepts `string|OpenSSLCertificate` and will return `OpenSSLCertificate` or `false`
 if invalid. So the same validation logic can be used as for strings?

Reply:
But if we already have an `OpenSSLCertificate`, we don't need to go through `openssl_x509_read`?